SrfGen/gcmt2srf.py

```diff
@@ -82,100 +87,210 @@ def run_create_srf(args, t, vs, rho, n_sims):
+            mwsr=args.mwsr,
+            seed=args.seed,
+            genslip_version=args.genslip_version,
+            silent=True,
+        )
+        srf_file = os.path.join(args.out_dir, pid, "Srf", "{}_REL01.srf".format(pid))
+        shypo, dhypo = focal_mechanism_2_finite_fault(
+            t.lat, t.lon, t.depth, t.mag, t.strike, t.rake, t.dip, args.mwsr
+        )[-2:]
+        gen_meta(
+            srf_file,
+            2,
+            t.mag,
+            t.strike,
+            t.rake,
+            t.dip,
+            args.dt,
+            lon=t.lon,
+            lat=t.lat,
+            vm=args.velocity_model,
+            shypo=shypo,
+            dhypo=dhypo,
+            file_name=os.path.join(args.out_dir, pid, pid),
+            mwsr=args.mwsr,
+        )
+    else:
+        stoch = os.path.join(args.out_dir, pid, "Stoch")
+        create_srf_psff(
+            t.lat,
+            t.lon,
+            t.mag,
+            t.strike,
+            t.rake,
+            t.dip,
+            t.depth,
+            args.dt,
+            prefix,
+            seed=args.seed,
+            stoch=stoch,
+            genslip_version=args.genslip_version,
+            mwsr=args.mwsr,
+        )
+    if args.plot:
+        call(["plot_srf_map.py", "%s.srf" % (prefix)])
 
-# load parameters
-parser = ArgumentParser()
-parser.add_argument("csv_file", help="path to CMT solutions CSV")
-parser.add_argument("velocity_model", help="path to 1D velocity model")
-parser.add_argument(
-    "-o", "--out-dir", help="directory to place outputs", default="./autosrf"
-)
-parser.add_argument("--dt", help="SRF timestep", type=float, default=0.005)
-parser.add_argument("-n", "--nproc", type=int, default=1)
-parser.add_argument("-p", "--plot", action="store_true")
-uncertainty_group = parser.add_argument_group(
-    "Uncertainty options",
-    "Options to be used when SRFs are to be genreated " "with uncertainty factors",
-)
-uncertainty_group.add_argument(
-    "-r", "--realistion-uncertainty", dest="uncertainty_file", type=str, default=""
-)
-uncertainty_group.add_argument(
-    "-c", "--cybershake-fault-file", dest="cs_file", type=str, default=""
-)
-args = parser.parse_args()
-# validate parameters
-if not os.path.exists(args.csv_file):
-    sys.exit("CMT solutions CSV file not found: %s" % (args.csv_file))
-if not os.path.exists(args.velocity_model):
-    sys.exit("Velocity model file not found: %s" % (args.velocity_model))
-all_opts = bool(args.uncertainty_file) and bool(args.cs_file)
-any_opts = bool(args.uncertainty_file) or bool(args.cs_file)
-if any_opts and not all_opts:
-    parser.error(
-        "If realisation uncertainty or a cybershake fault file is given then the other must be given also."
+
+def main():
+    # load parameters
+    parser = ArgumentParser(add_help=False)
+    parser.add_argument("csv_file", help="path to CMT solutions CSV")
+    parser.add_argument("velocity_model", help="path to 1D velocity model")
+    parser.add_argument(
```

Comment:
I don't think this is the functionality we want. @lee-robin Should it be FF for mw > 5.5 or how do we want to isolate doing type 2 generation vs type 1 generation.